crypto/s2n_ecc.c

```diff
@@ -28,16 +28,19 @@
 
 #define TLS_EC_CURVE_TYPE_NAMED 3
 
+/* IANA values can be found here: https://tools.ietf.org/html/rfc8446#appendix-B.3.1.4 */
+/* Share sizes are described here: https://tools.ietf.org/html/rfc8446#section-4.2.8.2
+ * and include the "legacy_form" byte */
 const struct s2n_ecc_named_curve s2n_ecc_supported_curves[2] = {
-    {.iana_id = TLS_EC_CURVE_SECP_256_R1, .libcrypto_nid = NID_X9_62_prime256v1, .name = "secp256r1"},
-    {.iana_id = TLS_EC_CURVE_SECP_384_R1, .libcrypto_nid = NID_secp384r1, .name= "secp384r1"},
+    {.iana_id = TLS_EC_CURVE_SECP_256_R1, .libcrypto_nid = NID_X9_62_prime256v1, .name = "secp256r1", .share_size = 65},
+    {.iana_id = TLS_EC_CURVE_SECP_384_R1, .libcrypto_nid = NID_secp384r1, .name= "secp384r1", .share_size = 97},
```

Comment:
I'd write out these magic numbers as (256 * 8) + 1 and (384 * 8) + 1 , just to make clear where they come from.  Or add a comment, or something just to note it :)